python/paddle/fluid/trainer.py

```diff
@@ -505,11 +505,21 @@ def _test_by_executor(self, reader, feed_order, fetch_list):
             accumulated = len(fetch_list) * [0]
             count = 0
             for data in reader():
-                outs = exe.run(program=self.test_program,
-                               feed=feeder.feed(data),
-                               fetch_list=fetch_list)
-                accumulated = [x[0] + x[1][0] for x in zip(accumulated, outs)]
-                count += 1
+                outs = None
+                try:
+                    outs = exe.run(program=self.test_program,
+                                   feed=feeder.feed(data),
+                                   fetch_list=fetch_list)
+                except core.EnforceNotMet as ex:
+                    print ex.message
+                except Exception as ex:
```

Comment:
I think we should let it crash if it really has some problems.

Reply:
But the crash message is not that helpful. 
`
paddle.fluid.core.EnforceNotMet: enforce begin_idx < end_idx failed, 9069 >= 9069
`
We can provide a better error message if empty input data is encountered.